File: python3/iniciante/1045.py
def sort(arr):
    for i in range(len(arr) - 1):
        for j in range(len(arr) - i - 1):
            if arr[j] < arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr

l = list(map(float, input().split()))

l = sort(l)
a, b, c = l[0], l[1], l[2]

if a >= b + c:
    print("NAO FORMA TRIANGULO")
else:
    if a ** 2 == b ** 2 + c ** 2:
        print("TRIANGULO RETANGULO")

    if a ** 2 > b ** 2 + c ** 2:
        print("TRIANGULO OBTUSANGULO")

    if a ** 2 < b ** 2 + c ** 2:
        print("TRIANGULO ACUTANGULO")

    if a == b == c:
        print("TRIANGULO EQUILATERO")

    if (a == b and a != c) or (a == c and a != b) or (b == c and b != a):
        print("TRIANGULO ISOSCELES")

File: c++/iniciante/1045.cpp
#include <iostream>
#include <algorithm>
#include <cmath>

int main() {
    float l[3];
    float a, b, c;
    std::cin >> l[0];
    std::cin >> l[1];
    std::cin >> l[2];

    std::sort(l, l + 3, std::greater<float>());
    a = l[0];
    b = l[1];
    c = l[2];

    if (a >= b + c) {
        std::cout << "NAO FORMA TRIANGULO" << std::endl;
        return 0;
    }

    if (std::pow(a, 2) == std::pow(b, 2) + std::pow(c, 2)) {
        std::cout << "TRIANGULO RETANGULO" << std::endl;
    }

    if (std::pow(a, 2) > std::pow(b, 2) + std::pow(c, 2)) {
        std::cout << "TRIANGULO OBTUSANGULO" << std::endl;
    }

    if (std::pow(a, 2) < std::pow(b, 2) + std::pow(c, 2)) {
        std::cout << "TRIANGULO ACUTANGULO" << std::endl;
    }

    if (a == b && b == c) {
        std::cout << "TRIANGULO EQUILATERO" << std::endl;
    }

    if ((a == b && a != c) || (a == c && a != b) || (b == c && b != a)) {
        std::cout << "TRIANGULO ISOSCELES" << std::endl;
    }

    return 0;
}
